dino-code/app/components/ChallengeCarousel.js

```diff
@@ -0,0 +1,318 @@
+    } else if (isRightSwipe && state.activeIndex > 0) {
+      dispatch({ type: "SET_ACTIVE_INDEX", payload: state.activeIndex - 1 });
+    }
+
+    // Reset values
+    touchStartX.current = null;
+    touchEndX.current = null;
+  };
+
+  // Mouse drag events (desktop support)
+  const handleMouseDown = (e) => {
+    dispatch({ type: "SET_IS_DRAGGING", payload: true });
+    dragStartX.current = e.clientX;
+  };
+
+  const handleMouseMove = (e) => {
+    if (state.isDragging) {
+      dragEndX.current = e.clientX;
+    }
+  };
+
+  const handleMouseUp = () => {
+    if (
+      state.isDragging &&
+      dragStartX.current !== null &&
+      dragEndX.current !== null
+    ) {
+      const distance = dragStartX.current - dragEndX.current;
+      const isLeftSwipe = distance > minSwipeDistance;
+      const isRightSwipe = distance < -minSwipeDistance;
+
+      if (isLeftSwipe && state.activeIndex < state.challenges.length - 1) {
+        dispatch({ type: "SET_ACTIVE_INDEX", payload: state.activeIndex + 1 });
+      } else if (isRightSwipe && state.activeIndex > 0) {
+        dispatch({ type: "SET_ACTIVE_INDEX", payload: state.activeIndex - 1 });
+      }
+    }
+
+    dispatch({ type: "SET_IS_DRAGGING", payload: false });
+    dragStartX.current = null;
+    dragEndX.current = null;
+  };
+
+  // Keyboard navigation
+  useEffect(() => {
+    const handleKeyDown = (e) => {
+      if (
+        e.key === "ArrowRight" &&
+        state.activeIndex < state.challenges.length - 1
+      ) {
+        dispatch({ type: "SET_ACTIVE_INDEX", payload: state.activeIndex + 1 });
+      } else if (e.key === "ArrowLeft" && state.activeIndex > 0) {
+        dispatch({ type: "SET_ACTIVE_INDEX", payload: state.activeIndex - 1 });
+      }
+    };
+
+    window.addEventListener("keydown", handleKeyDown);
+    return () => {
+      window.removeEventListener("keydown", handleKeyDown);
+    };
+  }, [state.activeIndex, state.challenges.length]);
+
+  // Function to conditionally join class names
+  const cn = (...classes) => {
+    return classes.filter(Boolean).join(" ");
+  };
+
+  return (
+    <div className={styles.carouselContainer}>
+      <div
+        className={styles.carouselWrapper}
+        onTouchStart={handleTouchStart}
+        onTouchMove={handleTouchMove}
+        onTouchEnd={handleTouchEnd}
+        onMouseDown={handleMouseDown}
+        onMouseMove={handleMouseMove}
+        onMouseUp={handleMouseUp}
+        onMouseLeave={handleMouseUp}
+        style={{ cursor: state.isDragging ? "grabbing" : "grab" }}
+      >
+        <div
+          className={styles.carouselTrack}
+          style={{ transform: `translateX(-${state.activeIndex * 100}%)` }}
+        >
+          {state.challenges.map((challenge) => (
+            <div key={challenge.id} className={styles.carouselSlide}>
+              <div className={styles.card}>
+                <div
+                  className={styles.cardHeader}
```

Comment:
use a pseudo element (::before or ::after) to apply the background card.https://developer.mozilla.org/en-US/docs/Web/CSS/::before